File: output/gemma4-e4b/avatar/Python/Java/codeforces_58_B.java
import java.util.Scanner;

class Solution {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        // Read initial n
        int n = scanner.nextInt();
        scanner.close();

        // Loop i from n down to 1
        for (int i = n; i >= 1; i--) {
            // Check divisibility using the current value of n
            if (n % i == 0) {
                // Update n
                n = i;
                // Print i followed by a space
                System.out.print(i + " ");
            }
        }
    }
}

File: output/gemma4-e4b/avatar/Java/Python/codeforces_58_B.py
import sys

def solve():
    # Read the single integer input n
    try:
        n = int(sys.stdin.read().strip())
    except:
        return

    # Print n followed by a space
    print(n, end=" ")

    m = n
    
    # Loop from n-1 down to 1 (inclusive)
    for i in range(n - 1, 0, -1):
        if m % i == 0:
            # Print i followed by a space
            print(i, end=" ")
            m = i

if __name__ == "__main__":
    solve()